contracts/ArrakisV2.sol

```diff
@@ -106,6 +120,46 @@ contract ArrakisV2 is IUniswapV3MintCallback, ArrakisV2Storage {
             token1.safeTransferFrom(msg.sender, me, amount1);
         }
 
+        if (isTotalSupplyGtZero) {
+            uint256 deposit0;
+            uint256 deposit1;
+            for (uint256 i; i < ranges.length; i++) {
+                Range memory range = ranges[i];
+                IUniswapV3Pool pool = IUniswapV3Pool(
+                    factory.getPool(
+                        address(token0),
+                        address(token1),
+                        range.feeTier
+                    )
+                );
+                uint128 liquidity = Position.getLiquidityByRange(
+                    pool,
+                    me,
+                    range.lowerTick,
+                    range.upperTick
+                );
+                if (liquidity == 0) continue;
+
+                liquidity = SafeCast.toUint128(
+                    FullMath.mulDiv(liquidity, mintAmount_, ts)
+                );
+
+                (uint256 amt0, uint256 amt1) = pool.mint(
+                    me,
+                    range.lowerTick,
+                    range.upperTick,
+                    liquidity,
+                    ""
+                );
+                deposit0 += amt0;
+                deposit1 += amt1;
+            }
+
+            /// TODO: this check may be provably unnecessary.
```

Comment:
In my view this can be removed and is provably safe. Rationale:

amount{0,1} are the amounts user deposited into arrakis by increasing current{0,1} by e.g. 10%

deposit{0,1} are the amount user deposits into uniswap by increasing all current liquidity by e.g. 10%

current{0,1} are derived by adding all current uniswap position liquidity AS WELL AS fees and leftover

This means that adding 10% to current{0,1} which results in amount{0,1} is STRICTLY greater than or equal to adding 10% to all uniswap ranges (since uniswap liq would always be STRICTLY less than or equal to current{0,1})